### ISSUE TITLE
Sentry types don't augment context app in Nuxt 2.13

### ISSUE THREAD

**Issue by rchl** [nuxt-community/sentry-module#195]
### Version

[v4.1.2](https://github.com/sentry-module/releases/tag/v4.1.2)

### Reproduction link

[https://codesandbox.io/s/still-brook-uy5cv?file=/plugins/sentry.js](https://codesandbox.io/s/still-brook-uy5cv?file=/plugins/sentry.js)

### Steps to reproduce

1. Open `plugins/sentry.js` and hover `app.$sentry` in the editor.

### What is expected ?

Should have proper Sentry types

### What is actually happening?

Has `any` type.

### Additional comments?

Works with `v0.7.9` of `@nuxt/types`

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/sentry-module/issues/c102">#c102</a>)</em></sub></div>

**Comment by kevinmarrec** [comment]
@rchl It sounds to be only reproducible on CodeSandBox, it works for me in VSCode :
![image](https://user-images.githubusercontent.com/25272043/85404304-0d6d2a00-b55f-11ea-82aa-76e951480b6e.png)

It probably won't output TS error during build as well, it's just CodeSandBox not preloading types correctly :/

**Comment by rchl** [comment]
I've initially reproduced it in my editor.

Also some more info from my investigation on Discord:

> I'm just getting `any` type for `app.$sentry` in 2.13
> So this works:
> `export { Context, Middleware, NuxtAppOptions, NuxtError, Plugin, Transition } from './app'`
> and this doesn't:
> `export * from './app'`
> in @nuxt/types/index.d.ts

**Comment by rchl** [comment]
I've updated the sandbox. Can you try again? What I want you to try is not necessarily VSCode itself as it has it's own language server for TS but just plain TS compilation.

I've modified the project by:
 - Adding `compile` script
 - Adding `include` array with `plugins/sentry.js` in `tsconfig.json`
 - Changing the sentry.js to expose issues due to using `any` type.

Just get local copy of the project and run `yarn compile` on it. You should see a compilation issue like:
```
yarn compile                                                                                                                                              1230  14:48:02   
yarn run v1.22.4                                                                                                                                                                   
$ tsc                                                                                                                                                                              
plugins/sentry.js:3:25 - error TS7006: Parameter 'scope' implicitly has an 'any' type.

3   app.$sentry.withScope(scope => scope.clear())
                          ~~~~~


Found 1 error.
```

**Comment by rchl** [comment]
Update. Actually, it works fine after adding latest `typescript` dependency to the project.
So the one that yarn ran for me must have been older.

**Comment by kevinmarrec** [comment]
@rchl It's not the same issue here, your last issue is cause of TypeScript applying `strict` check (you can try to disable `strict` in `tsconfig.json`) on JavaScript files, which has probably been fixed in `3.9`.

**Comment by kevinmarrec** [comment]
FYI : I still got lof of unfixable issues with TypeScript 3.9 so i'm still not using in projects for now.

**Comment by rchl** [comment]
> your last issue is cause of TypeScript applying strict any check

No, the error would not trigger had the types be applied properly.

With latest typescript the types are known, both when compiling and in my editor (ST).

So I'm fine with resolving it as long as you are fine with `@nuxt/types` requiring latest TS version.

**Comment by kevinmarrec** [comment]
> So I'm fine with resolving it as long as you are fine with @nuxt/types requiring latest TS version.
https://github.com/nuxt-community/sentry-module/issues/195#issuecomment-648135364

**Comment by rchl** [comment]
Well, then should the `@nuxt/types` be changed to not use the `export *` form?

The compilation fails with TS 3.8.3 and doesn't fail with TS 3.9.2.

**Comment by rchl** [comment]
I suppose the fix that makes it work is this one:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#export--is-always-retained

> In previous TypeScript versions, declarations like export * from "foo" would be dropped in our JavaScript output if foo didn’t export any values. This sort of emit is problematic because it’s type-directed and can’t be emulated by Babel. TypeScript 3.9 will always emit these export * declarations. In practice, we don’t expect this to break much existing code.

I guess that's the case here since `@nuxt/types` doesn't export any values, just types.

**Comment by kevinmarrec** [comment]
@rchl I'm not able to reproduce here : https://github.com/kevinmarrec/sentry-repro

**Comment by rchl** [comment]
@kevinmarrec missing `"checkJs": true,`

**Comment by kevinmarrec** [comment]
@rchl Indeed, well I think we probably need to export each interface instead of `export *`, like it was done before, so it keeps compatibility with TS 3.8

**Comment by pi0** [comment]
Hi! This should be fixed with [v2.13.1](https://github.com/nuxt/nuxt.js/releases/tag/v2.13.1).
